nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/util/DataTypeUtils.java

```diff
@@ -1918,6 +1918,7 @@ public static DataType getDataTypeFromSQLTypeValue(final int sqlType) {
             case Types.BIGINT:
                 return RecordFieldType.BIGINT.getDataType();
             case Types.BOOLEAN:
+            case Types.BIT:
```

Comment:
The mapping for the `BIT` datatype should be changed to `Integer`. This is because the `BIT` datatype can represent not only BIT(1) but also BIT(n>1), making it impossible to distinguish between them. Therefore, using `Integer` provides a more suitable and consistent mapping.

Reply:
Agreed

Reply:
hi @Lehel44 and @mattyb149

Have a look on below  sample table definition (Postgres14)

    CREATE TABLE test_table (
        id int4 NOT NULL,
        active bool NULL,
        col_varbit varbit(2) NULL,
        col_bit bit(1) NULL,
        CONSTRAINT temp_market_cap_pkey PRIMARY KEY (id)
    );



| colName     | postgresType | javaSqlType | finalType |
|-------------|--------------|-------------|-----------|
| id          | int          | 4           | INT       |
| active      | bool         | -7          | BOOLEAN   |
| col_bit     | bit(1)       | -7          | BOOLEAN   |
| col_varbit  | varbit(2)    | 1111        | STRING    |

So I believe that we are doing right ,let me know yours thought

References: https://www.postgresql.org/message-id/Pine.BSO.4.61.0511231340310.14387%40leary.csoft.net

Reply:
hi @Lehel44 and @mattyb149

Have a look on below  sample table definition (Postgres14)

    CREATE TABLE test_table (
        id int4 NOT NULL,
        active bool NULL,
        col_bit bit(1) NULL,
        col_bit1 bit(1) NULL,
        col_varbit varbit(2) NULL,
        CONSTRAINT temp_market_cap_pkey PRIMARY KEY (id)
    );

We can use something like below:

    String typeName = columnResultSet.getString("TYPE_NAME");
    final int dataType;
    if (typeName.equalsIgnoreCase("bool")) {
        dataType = 16;
    } else {
        dataType = columnResultSet.getInt("DATA_TYPE");
    }
            
then below will be result 

| colName    | postgresType | javaSqlType | javaSqlTypeName | finalType |
|------------|--------------|-------------|-----------------|-----------|
| id         | int          | 4           | int4            | INT       |
| active     | bool         | -7          | bool            | BOOLEAN   |
| col_bit    | bit(1)       | -7          | bit             | INT       |
| col_bit1   | bit(2)       | -7          | bit             | INT       |
| col_varbit | varbit(2)    | 1111        | varbit          | STRING    |

let me know your thoughts

Reply:
@ravinarayansingh I tried it and it looks good to me!